### PR TITLE
Bundle and minimize CSS & JS assets

### PR DESCRIPTION
**Pull request by patrickod**
First step in addressing freedomofpress/securedrop#1152

Uses flask-assets to group the CSS assets, run them through
cssmin, and link to them in the index.html and base.html templates
for both journalists and sources

Adds both flask-assets and cssmin as securedrop depedencies

### CODE REVIEW

**Comment by patrickod** [comment]
Derp there's an issue with relative paths in the `font-awesome.css` file pointing to the font files, whereby the minimized CSS uses broken links and the font 404s :(

**Comment by patrickod** [comment]
Path issues are now fixed. Have tested this using onion services locally.

This PR also sets inline height & width attributes on the logo `img` tags in both the source and document interfaces. This prevents the logo image redrawing when the stylesheet is loaded async

**Comment by patrickod** [comment]
New commit adds the asset working directories in the `.gitignore`

**Comment by garrettr** [comment]
Because of the various issues with the current build environment (described in https://github.com/freedomofpress/securedrop/issues/1152#issuecomment-149355478), I'm documenting my testing procedure here for the benefit of future generations:

1. Build the OSSEC debs. Since our goal is to install locally built debs instead of the debs from the production apt server, we need to build _all_ of the debs needed by SecureDrop.
    - [ ] As we rework the `install_local_pkgs` role in the future, it might be good to be able to only install a subset of the required debs from local.
    - `git clone https://github.com/freedomofpress/ossec`
    - `cd ossec` and `vagrant up build` (or `vagrant provision build` to re-build the debs if you've already run `up`). This runs an Ansible playbook which builds Ubuntu-compatible .debs of the upstream OSSEC.
    - When the playbook finishes, the debs will be in the `build` subdirectory. `cp` the debs to the `build` subdirectory of your **securedrop** git repo.
2. Build the SecureDrop debs.
    - `cd` into your securedrop repo and run `vagrant up build` (or `vagrant provision build` if the VM already exists) to similarly build all of the debs.
3. **Awful hack alert**: Comment out line 83 from the `Vagrantfile`: `ansible.skip_tags = ENV['SECUREDROP_STAGING_SKIP_TAGS'] || 'install_local_pkgs'`
4. Run `vagrant destroy -f /staging/ && vagrant up /staging/`
5. Since we're interested in testing the perceived performance difference of the interfaces when they are served as Tor Hidden Services, we need to be able to access them.
    - If you want to access the document interface, you will need to add a `HidServAuth` line to your torrc with the value from `install_files/ansible-base/app-document-aths`.
6. Open your Tor-proxied browser and visit the hidden services from `app-source-ths` and `app-document-aths` to view the changes.

File: securedrop/source_templates/index.html [hunk 1]
```diff
@@ -17,7 +20,7 @@
     <div class="content">
       <div class="grid">
         <div id="header">
-          <img src="/static/i/{{ header_image }}" class="logo grid-item" alt="SecureDrop">
+          <img src="/static/i/{{ header_image }}" class="logo grid-item" alt="SecureDrop" width="150px" height="135px">
```
Comment (garrettr) [inline securedrop/source_templates/index.html]: The width/height are wrong, and the image looks distorted as a result:

screen shot 2015-11-08 at 1 09 26 pm

According to `source.css`, width should be 250px. Height should be whatever maintains the aspect ratio to that width.

Note that the actual `logo.png` has dimensions that are twice as large as this (500x450 px, documented [here](https://docs.securedrop.org/en/latest/install.html#configure-the-installation)). This is intentional, so the web interfaces look good on computers with HiDPI displays (e.g. Apple Retina).

Comment (patrickod) [inline securedrop/source_templates/index.html]: ah! nice catch. will update

**Comment by patrickod** [comment]
Updated with corrected `img` tag attributes

**Comment by garrettr** [comment]
Rebased and merged into develop in e74b39075fa4d407bf929ab8bc5ab24abad79f70, e4cf8c8d6735ab36fdb55a33d9a7cadc359cd1bf, c0fa4b35cd72e8cd81e05b3cf1a3b0b19cf25e72, 2617b7effc86d074d0bf489e515aa7a22a33b8da.

Fixed a missing comma and incorrect minified CSS path in the assets tag for the Source Interface base template in 5b9f6696c0c27a42791aeaa43afc99e76b5c4a06.

**Pull request closed without merging**